lib/logstash/gemfile.rb

```diff
@@ -0,0 +1,173 @@
+module LogStash
```

Comment:
this is really a first draft implementation, I'd prefer to keep it like that for now. these 3 classes are really meant to work together and when we'll improve on it we will certainly have to refactor some parts and then we can decide to split?